### PR TITLE
UI - Add Network: shows "Offering for Isolated networks with no Source Nat service" on Network Offering for normal users

### PR DESCRIPTION
**Pull request by utchoang**
### Description

This PR fixes https://github.com/apache/cloudstack/issues/5897

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by sureshanaparti** [comment]
@blueorangutan ui

File: ui/src/views/network/CreateIsolatedNetworkForm.vue [hunk 1]
```diff
@@ -344,11 +344,13 @@ export default {
       var params = {
         zoneid: this.selectedZone.id,
         guestiptype: 'Isolated',
-        supportedServices: 'SourceNat',
         state: 'Enabled'
       }
-      if (isAdminOrDomainAdmin() && this.selectedDomain.id !== '-1') { // domain is visible only for admins
-        params.domainid = this.selectedDomain.id
+      if (isAdminOrDomainAdmin()) {
+        params.supportedServices = 'SourceNat'
```
Comment (sureshanaparti) [inline ui/src/views/network/CreateIsolatedNetworkForm.vue:350]: @utchoang this is applicable for normal users as they can only create isolated networks with source nat service.

Comment (utchoang) [reply ui/src/views/network/CreateIsolatedNetworkForm.vue:350]: @sureshanaparti Does this mean that this change is not correct?

Comment (utchoang) [reply ui/src/views/network/CreateIsolatedNetworkForm.vue:350]: Does it mean for normal users and admin can only add isolated networks with source nat service?

Comment (sureshanaparti) [reply ui/src/views/network/CreateIsolatedNetworkForm.vue:350]: normal users can add isolated networks with only source nat service

Comment (utchoang) [reply ui/src/views/network/CreateIsolatedNetworkForm.vue:350]: Maybe I'll temporarily draft this PR and wait for discussion?

Comment (sureshanaparti) [reply ui/src/views/network/CreateIsolatedNetworkForm.vue:350]: Hi @utchoang Can you update the changes to enable only admin to view the network offerings without sourcenat. Users can see offerings with sourcenat enabled only.

**Comment by weizhouapache** [comment]
this depends on #5920 . when #5920 is approved and merged into 4.16/main, I will review this pr.

**Comment by DaanHoogland** [comment]
> this depends on #5920 . when #5920 is approved and merged into 4.16/main, I will review this pr.

it just was @weizhouapache

**Comment by sureshanaparti** [comment]
> > this depends on #5920 . when #5920 is approved and merged into 4.16/main, I will review this pr.
> 
> it just was @weizhouapache

@DaanHoogland @weizhouapache work in this PR is still in progress. don't review now.

File: ui/src/views/network/CreateIsolatedNetworkForm.vue (hunk 1 above)
Comment (sureshanaparti) [reply ui/src/views/network/CreateIsolatedNetworkForm.vue:350]: @utchoang Are the changes done, and is this PR ready?

**Comment by sureshanaparti** [comment]
Hi @utchoang Is this PR ready for review / testing.

**Comment by sureshanaparti** [comment]
@blueorangutan ui

**Comment by sureshanaparti** [comment]
Verified with admin and user accounts, able to view the network offerings without source nat. LGTM.

**Review by sureshanaparti — APPROVED** [review]

**Comment by weizhouapache** [comment]
I have tested it just now.

root admin is able to see network offerings for isolated networks, vpc networks (with specifyvlan=true or false). good.
normal user  is able to see network offerings for isolated networks, vpc networks (with specifyvlan=false). good
However... 
domain admin can only see network offerings for isolated networks, but not vpc networks.

**Comment by utchoang** [comment]
@blueorangutan ui

**Comment by sureshanaparti** [comment]
@blueorangutan ui

**Comment by sureshanaparti** [comment]
> I have tested it just now.
> 
> root admin is able to see network offerings for isolated networks, vpc networks (with specifyvlan=true or false). good. normal user is able to see network offerings for isolated networks, vpc networks (with specifyvlan=false). good However... domain admin can only see network offerings for isolated networks, but not vpc networks.

Hi @weizhouapache can you confirm the domain admin case with the latest changes.

**Comment by weizhouapache** [comment]
@utchoang @sureshanaparti 
I am still not able to see offering for vpc networks when login as domain admin.

see screenshot below
image


for normal users, it looks ok. see below
image

**Comment by weizhouapache** [comment]
after discussion with @sureshanaparti , I checked my environment and found out it is because regular user has 1 vpc but domain admin does not. it makes sense.

**Review by weizhouapache — APPROVED** [review]

**Comment by sureshanaparti** [comment]
thanks for testing this @weizhouapache

**Pull request merged**